src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -33,8 +37,14 @@ public LogicManager(Model model, Storage storage) {
     public CommandResult execute(String commandText) {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
         Command command = parser.parseCommand(commandText);
+
         command.setData(model);
-        return command.execute();
+        CommandResult result = command.execute();
+
+        if (model.hasUncommittedChanges() && !(command instanceof UndoCommand) && !(command instanceof RedoCommand) ) {
```

Comment:
The `command instanceof UndoCommand` etc. is just papering over bugs in your `ModelManager` code. You should not need it if your implementation is correct.